sw/host/opentitantool/src/command/lc.rs

```diff
@@ -54,8 +56,87 @@ impl CommandDispatch for LcStateRead {
     }
 }
 
+#[derive(Debug, Args)]
+/// Initiates a device transition from Raw to TestUnlocked0.
+pub struct RawUnlock {
+    /// The raw unlock token hexstring.
+    #[arg(long, default_value = "0x00000000_00000000_00000000_00000000")]
+    pub token: String,
+
+    /// Reset duration when switching the LC TAP straps.
+    #[arg(long, value_parser = parse_duration, default_value = "100ms")]
+    pub reset_delay: Duration,
+
+    #[command(flatten)]
+    pub jtag_params: JtagParams,
+}
+
+impl CommandDispatch for RawUnlock {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        // Set the TAP straps for the lifecycle controller and reset.
+        transport.pin_strapping("PINMUX_TAP_LC")?.apply()?;
+        transport.reset_target(self.reset_delay, true)?;
+
+        // Spawn an OpenOCD process and connect to the LC JTAG TAP.
+        let jtag = self.jtag_params.create(transport)?;
+        jtag.connect(JtagTap::LcTap)?;
+
+        // Read, decode, and check the LC state is Raw.
+        let lc_state = DifLcCtrlState(jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?);
+        if lc_state != DifLcCtrlState::Raw {
+            return Err(anyhow!(
+                "Device must be in Raw LC state. It is in: {}",
+                lc_state
+            ));
+        }
+
+        // Parse unlock token.
+        let hex_str_no_sep = self.token.replace('_', "");
+        let hex_str_prefix = "0x";
+        let sanitized_hex_str = if self.token.starts_with(hex_str_prefix) {
+            hex_str_no_sep.strip_prefix(hex_str_prefix).unwrap()
+        } else {
+            hex_str_no_sep.as_str()
+        };
+        let token_bytes_vec = decode(sanitized_hex_str)?;
+        let token_word_vec: Vec<u32> = token_bytes_vec
+            .chunks_exact(std::mem::size_of::<u32>())
+            .map(|chunk| u32::from_be_bytes(chunk.try_into().unwrap()))
+            .rev()
+            .collect();
+        if token_word_vec.len() < 4 {
```

Comment:
This check needs to be moved before `token_word_vec` so that if the input is 17 bytes, the extra byte is not silently ignored.

Reply:
ah good catch. fixed.